sys/userlib/src/macros.rs

```diff
@@ -33,7 +33,18 @@ macro_rules! sys_log {
     };
 }
 
-#[cfg(not(any(feature = "log-semihosting", feature = "log-itm")))]
+#[cfg(feature = "log-null")]
+#[macro_export]
+macro_rules! sys_log {
+    ($s:expr) => {};
+    ($s:expr, $($tt:tt)*) => {};
+}
+
+#[cfg(not(any(
```

Comment:
YMMV, but whenever I see both positive and negative `#[cfg...` branches, I prefer to replace it with a `cfg_if` block (and then use `else` for the negative case).

Reply:
Heh, amusingly, I did this for a _lot_ of cases in this changeset, but not this one. Fixing.